src/agent/border_agent.cpp

```diff
@@ -250,61 +288,74 @@ void BorderAgent::StartPublishService(void)
     {
         mPublisher->Start();
     }
+
+exit:
+    return;
 }
 
 void BorderAgent::StopPublishService(void)
 {
-    otbrLog(OTBR_LOG_INFO, "Stop publishing service");
+    VerifyOrExit(mPublisher != NULL);
 
     if (mPublisher->IsStarted())
     {
         mPublisher->Stop();
     }
+
+exit:
+    otbrLog(OTBR_LOG_INFO, "Stop publishing service");
 }
 
-void BorderAgent::HandleThreadChange(void)
+void BorderAgent::SetNetworkName(const char *aNetworkName)
 {
-    otbrLog(OTBR_LOG_INFO, "Handle Thread change");
+    strncpy(mNetworkName, aNetworkName, sizeof(mNetworkName) - 1);
 
-    if (mThreadStarted && mNetworkName[0] != '\0')
+#if OTBR_ENABLE_MDNS_AVAHI || OTBR_ENABLE_MDNS_MDNSSD
```

Comment:
is it necessary to have this `#if`? as either `avahi` or `mDNSResponder` would be used, it is always true.